BHoM_Engine/Modify/AddFragment.cs

```diff
@@ -0,0 +1,64 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+using BH.oM.Base;
+using BH.oM.Reflection.Attributes;
+using System.ComponentModel;
+
+namespace BH.Engine.Base
+{
+    public static partial class Modify
+    {
+        [Description("Appends a Fragment Property to a given BHoM Object")]
+        [Input("iBHoMObject", "Any object implementing the IBHoMObject interface that can have fragment properties appended to it")]
+        [Input("fragment", "Any fragment object implementing the IBHoMFragment interface to append to the object")]
+        [Input("replace", "If true will replace exisiting fragments ")]
+        [Output("iBHoMObject", "The BHoM object with the added fragment")]
+        public static IBHoMObject AddFragment(this IBHoMObject iBHoMObject, IBHoMFragment fragment, bool replace = false)
+        {
+            if (iBHoMObject == null) return null;
+            IBHoMObject o = iBHoMObject.DeepClone();
+            o.Fragments = new List<IBHoMFragment>(iBHoMObject.Fragments);
+            if(o.Fragments.Contains(fragment))
+            {
+                if(replace)
+                {
+                    o.Fragments[o.Fragments.IndexOf(fragment)] = fragment;
+                }
+                else
+                {
+                    Reflection.Compute.RecordError("Fragment already exists");
```

Comment:
Can we expand this error message to tell the user something like "That fragment already exists on this object. If you would like to replace the existing fragment set the 'replace' input to 'true'"

Reply:
done